src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -739,6 +740,12 @@ deparse_shard_index_statement(IndexStmt *origStmt, Oid distrelid, int64 shardid,
 												relationName),
 					 indexStmt->accessMethod);
 
+	/* switch to search_path = pg_catalog before deparsing index columns */
```

Comment:
maybe we can also have some more comments as in `GetForeignConstraintCommandsToReferenceTable`

```c
	/*
	 * Set search_path to NIL so that all objects outside of pg_catalog will be
	 * schema-prefixed. pg_catalog will be added automatically when we call
	 * PushOverrideSearchPath(), since we set addCatalog to true;
	 */
```

Reply:
or we can just use PushOverrideEmptySearchPath